build/debian-iptables/iptables-wrapper.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+# Detect whether the base system is using iptables-legacy or
+# iptables-nft. This assumes that some non-containerized process (eg
+# kubelet) has already created some iptables rules.
+num_legacy_lines=$( (iptables-legacy-save || true; ip6tables-legacy-save || true) 2>/dev/null | wc -l)
```

Comment:
what about not ip6 enabled systems?  Will this error?

Reply:
yes; that's what the "`|| true`" is for.
I'm continuing to work on https://github.com/danwinship/iptables-wrappers/ and one of the iptables hackers suggested we should add "` | grep -v "^\([*:#]\|COMMIT"`" here too so we're only comparing actual rules and don't get tricked if there are just lots of tables. (Although I think that's probably not really a problem since we know kubelet will create like 8 rules or so before this script ever runs, so even if all the default tables exist in the "wrong" version, we should still be OK.)

Reply:
any reason _not_ to check for actual rules?